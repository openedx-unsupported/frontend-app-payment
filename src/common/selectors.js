import { createSelector } from 'reselect';

export const userAccountSelector = state => state.userAccount;

export const emailSelector = createSelector(
  userAccountSelector,
  userAccount => userAccount.email,
);

export const configurationSelector = state => state.configuration;
