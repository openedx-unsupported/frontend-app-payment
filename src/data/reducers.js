import { combineReducers } from 'redux';
import { reducer as formReducer } from 'redux-form';

import {
  reducer as paymentReducer,
  storeName as paymentStoreName,
} from '../payment';
import {
  reducer as feedbackReducer,
  storeName as feedbackStoreName,
} from '../feedback';

const createRootReducer = () => combineReducers({
  [paymentStoreName]: paymentReducer,
  [feedbackStoreName]: feedbackReducer,
  form: formReducer,
});

export default createRootReducer;
