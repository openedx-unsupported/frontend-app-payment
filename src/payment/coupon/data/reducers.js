import { ADD_COUPON, REMOVE_COUPON, UPDATE_COUPON_DRAFT } from './actions';

const defaultState = {
  benefit: null,
  code: null,
  voucherId: null,
  error: null,
  loaded: false,
  loading: false,
};

const reducer = (state = defaultState, action = null) => {
  if (action !== null) {
    switch (action.type) {
      case ADD_COUPON.BEGIN:
        return {
          ...state,
          loading: true,
          loaded: false,
          error: null,
        };
      case ADD_COUPON.SUCCESS:
        return {
          ...state,
          loading: false,
          loaded: true,
          error: null,
          benefit: action.payload.benefit,
          code: action.payload.code,
          voucherId: action.payload.voucherId,
        };
      case ADD_COUPON.FAILURE:
        return {
          ...state,
          loading: false,
          loaded: false,
          error: action.payload.error,
          voucherId: null,
        };
      case REMOVE_COUPON.BEGIN:
        return {
          ...state,
          loading: true,
          loaded: false,
          error: null,
        };
      case REMOVE_COUPON.SUCCESS:
        return defaultState;
      case REMOVE_COUPON.FAILURE:
        return {
          ...state,
          loading: false,
          loaded: true, // There's still a loaded coupon.
          error: action.payload.error,
        };
      case UPDATE_COUPON_DRAFT:
        return {
          ...state,
          code: action.payload.code,
        };
      default:
    }
  }

  return state;
};

export default reducer;
