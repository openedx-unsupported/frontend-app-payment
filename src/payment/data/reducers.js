import { combineReducers } from 'redux';

import {
  BASKET_DATA_RECEIVED,
  BASKET_PROCESSING,
  CAPTURE_KEY_DATA_RECEIVED,
  CAPTURE_KEY_PROCESSING,
  MICROFORM_STATUS,
  fetchBasket,
  submitPayment,
  fetchCaptureKey,
} from './actions';

import { DEFAULT_STATUS } from '../checkout/payment-form/flex-microform/constants';

const basketInitialState = {
  loading: true,
  loaded: false,
  submitting: false,
  redirect: false,
  isBasketProcessing: false,
  products: [],
};

const basket = (state = basketInitialState, action = null) => {
  if (action !== null) {
    switch (action.type) {
      case fetchBasket.TRIGGER: return { ...state, loading: true };
      case fetchBasket.FULFILL: return {
        ...state,
        loading: false,
        loaded: true,
      };

      case BASKET_DATA_RECEIVED: return { ...state, ...action.payload };

      case BASKET_PROCESSING: return {
        ...state,
        isBasketProcessing: action.payload,
      };

      case submitPayment.TRIGGER: return {
        ...state,
        paymentMethod: action.payload.method,
      };
      case submitPayment.REQUEST: return {
        ...state,
        submitting: true,
      };
      case submitPayment.SUCCESS: return {
        ...state,
        redirect: true,
      };
      case submitPayment.FULFILL: return {
        ...state,
        submitting: false,
        paymentMethod: undefined,
      };

      default:
    }
  }
  return state;
};

const captureContextInitialState = {
  isCaptureKeyProcessing: false,
  microformStatus: DEFAULT_STATUS,
  captureKeyId: '',
};

const captureKey = (state = captureContextInitialState, action = null) => {
  if (action !== null) {
    switch (action.type) {
      case fetchCaptureKey.TRIGGER: return state;
      case fetchCaptureKey.FULFILL: return state;

      case CAPTURE_KEY_DATA_RECEIVED: return { ...state, ...action.payload };

      case CAPTURE_KEY_PROCESSING: return {
        ...state,
        isCaptureKeyProcessing: action.payload,
      };

      case MICROFORM_STATUS: return {
        ...state,
        microformStatus: action.payload,
      };

      default:
    }
  }
  return state;
};

const reducer = combineReducers({
  basket,
  captureKey,
});

export default reducer;
