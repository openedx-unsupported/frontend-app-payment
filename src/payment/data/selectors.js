import { createSelector } from 'reselect';
import {
  localeSelector,
  getCountryList,
} from '@edx/frontend-i18n';

export const storeName = 'payment';
export const paymentSelector = (state) => {
  const paymentState = state[storeName].basket;
  const loaded = !paymentState.loading && !paymentState.loadingError;
  return {
    ...paymentState,
    loaded,
    isEmpty: loaded && paymentState.products && paymentState.products.length > 0,
  };
};

export const countryOptionsSelector = createSelector(
  localeSelector,
  locale => ({
    countryOptions: getCountryList(locale).map(({ code, name }) => ({ value: code, label: name })),
  }),
);

export const basketSelector = state => ({ ...state[storeName].basket });

export const productsSelector = state => ({ ...state[storeName].basket });
