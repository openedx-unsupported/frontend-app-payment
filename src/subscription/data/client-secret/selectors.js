import { createSelector } from 'reselect';
import { storeName } from '../constants';

export const clientSecretSelector = state => (state[storeName] ? state[storeName].clientSecret : null);

export const getClientSecretSelector = createSelector(
  clientSecretSelector,
  (clientSecret, isClientSecretProcessing) => ({
    clientSecretId: clientSecret && clientSecret.capture_context ? clientSecret.capture_context.key_id : null,
    isClientSecretProcessing,
  }),
);
