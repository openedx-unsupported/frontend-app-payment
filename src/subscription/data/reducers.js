import { combineReducers } from 'redux';

import subscription from './details/reducer';
import clientSecret from './client-secret/reducer';

export const reducer = combineReducers({
  details: subscription,
  clientSecret,
});

export default reducer;
