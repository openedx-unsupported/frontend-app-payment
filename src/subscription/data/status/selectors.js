// import { createSelector } from 'reselect';
import { storeName } from '../constants';

// eslint-disable-next-line import/prefer-default-export
export const subscriptionStatusSelector = state => state[storeName].status;

// export const status = createSelector(
//   subscriptionStatusSelector,
//   status => status,
// );
