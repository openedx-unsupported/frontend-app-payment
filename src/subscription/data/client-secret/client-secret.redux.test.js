import { createStore, combineReducers } from 'redux';

import { reducer } from '../reducers';
import {
  clientSecretDataReceived,
  clientSecretProcessing,
} from './actions';
import { clientSecretSelector } from './selectors';

jest.mock('universal-cookie', () => {
  class MockCookies {
    static result = {};

    get() {
      return MockCookies.result;
    }
  }
  return MockCookies;
});

describe('subscription clientSecret redux tests', () => {
  let store;

  beforeEach(() => {
    store = createStore(combineReducers({
      subscription: reducer,
    }));
  });

  describe('selectors', () => {
    // describe('details', () => {
    it('default state', () => {
      const result = clientSecretSelector(store.getState());
      expect(result).toEqual({
        isClientSecretProcessing: false,
        clientSecretId: '',
      });
    });
  });

  describe('clientSecret reducer', () => {
    it('should return the default state when appropriate', () => {
      // Its base state
      expect(store.getState().subscription.clientSecret).toMatchSnapshot();

      // If an action isn't appropriate for it
      store.dispatch({ type: 'SOMETHING_ELSE' });
      expect(store.getState().subscription.clientSecret).toMatchSnapshot();

      // When called with no parameters
      expect(reducer()).toMatchSnapshot();
    });

    it('SUBSCRIPTION_CLIENT_SECRET_DATA_RECEIVED action', () => {
      store.dispatch(clientSecretDataReceived({ capture_context: { key_id: 'bar' } }));
      expect(store.getState().subscription.clientSecret.clientSecretId).toBe('bar');
    });

    describe('SUBSCRIPTION_CLIENT_SECRET_PROCESSING action', () => {
      it('SUBSCRIPTION_CLIENT_SECRET_PROCESSING true action', () => {
        store.dispatch(clientSecretProcessing(true));
        expect(store.getState().subscription.clientSecret.isClientSecretProcessing).toBe(true);
      });

      it('SUBSCRIPTION_CLIENT_SECRET_PROCESSING false action', () => {
        store.dispatch(clientSecretProcessing(false));
        expect(store.getState().subscription.clientSecret.isClientSecretProcessing).toBe(false);
      });
    });
  });
});
