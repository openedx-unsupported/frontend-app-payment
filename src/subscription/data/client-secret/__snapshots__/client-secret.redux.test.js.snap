// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`subscription clientSecret redux tests clientSecret reducer should return the default state when appropriate 1`] = `
Object {
  "clientSecretId": "",
  "isClientSecretProcessing": false,
}
`;

exports[`subscription clientSecret redux tests clientSecret reducer should return the default state when appropriate 2`] = `
Object {
  "clientSecretId": "",
  "isClientSecretProcessing": false,
}
`;

exports[`subscription clientSecret redux tests clientSecret reducer should return the default state when appropriate 3`] = `
Object {
  "clientSecret": Object {
    "clientSecretId": "",
    "isClientSecretProcessing": false,
  },
  "details": Object {
    "certificateType": "verified",
    "currency": "USD",
    "isSubscriptionDetailsProcessing": false,
    "loaded": false,
    "loading": true,
    "organization": "University of California, Berkeley",
    "paymentMethod": "stripe",
    "price": 49,
    "products": Array [],
    "programTitle": "Blockchain Fundamentals",
    "redirect": false,
    "submitting": false,
  },
}
`;
